sound/soc/sof/intel/hda-dsp.c

```diff
@@ -295,6 +296,12 @@ static int hda_suspend(struct snd_sof_dev *sdev, int state)
 	hda_dsp_ipc_int_disable(sdev);
 
 #if IS_ENABLED(CONFIG_SND_SOC_SOF_HDA)
+	if (IS_ENABLED(CONFIG_SND_SOC_SOF_HDA_AUDIO_CODEC) && runtime_suspend)
+		/* enable controller wake up event */
+		snd_hdac_chip_writew(bus, WAKEEN,
+				     snd_hdac_chip_readw(bus, WAKEEN) |
+				     STATESTS_INT_MASK);
```

Comment:
Question: what happens if we have HDA_AUDIO_CODEC selected but only have HDMI on the platform, e.g. on Up2 with no I2S codec. Does this cause any problems?

Reply:
yes, It causes a unexpected interrupt when resuming from suspend mode and makes IPC timeout and system hang. I am working on it.

Reply:
In my branch I modified this your patch to do
```
	snd_hdac_chip_writew(bus, WAKEEN,
			     snd_hdac_chip_readw(bus, WAKEEN) |
			     bus->codec_mask);
```
That's a bit better than enabling all interrupt sources, but still not perfect. Maybe we should exclude HDMI codecs too.

Reply:
yes, I have a patch for another feature and a codec mask for only hda codec will be introduced.